src/codec/framed_write.rs

```diff
@@ -267,23 +268,18 @@ impl<T, B> FramedWrite<T, B> {
     }
 }
 
-impl<T: io::Read, B> io::Read for FramedWrite<T, B> {
-    fn read(&mut self, dst: &mut [u8]) -> io::Result<usize> {
-        self.inner.read(dst)
-    }
-}
-
-impl<T: AsyncRead, B> AsyncRead for FramedWrite<T, B> {
-    fn read_buf<B2: BufMut>(&mut self, buf: &mut B2) -> Poll<usize, io::Error>
```

Comment:
We also want to forward `poll_read_buf`, to allow for gathering reads.

Reply:
Done